### DISCUSSION TITLE
Raspberry Pi 5

### DISCUSSION THREAD

**Discussion by probonopd** [probonopd/MiniDexed#556] [General]
Have a Raspberry Pi 5 incoming, should arrive tomorrow. However, [the Circle documentation](https://github.com/rsta2/circle) currently says

> The Raspberry Pi 5 is also not supported yet. 

@rsta2 let me know if I can help with testing in any way.

**Comment by pik33** [comment]
I now have one. It starts fast with Linux (<20s to desktop with SD, <15s to desktop with USB SSD). If configuring the linux to be minimal these times may be even shorter.

This means maybe it is worth converting minidexed to use linux instead of bare metal for RPi5 until any bare metal environment can even be compatible with 5. The hardware changes are huge, now all I/O stuff is done in the RP1 chip and not the SoC itself.

**Reply by probonopd** [reply]
Interesting that you are saying this. I was wondering the same when my RPi 5 arrived and was able to boot into a full desktop so quickly (using a V30 marked microSD).

Imagine we could:
* Build the same source code against the Linux kernel instead of against bare-metal Circle
* Make a minial custom Linux system (like LibeELEC) that is "just enough operating system" to run that one binary

That woud require Circle to be converted into something that can be built into Linux executables as well as bare metal kernels. Wdyt @rsta2? It'd certainly be a lot of work, but adding bare-metal support for the RPi 5 is probably going to be a lot of work too. 

Advantages would be:
* Building binaries that can run on the Linux kernel would possibly allow Circle to be ported to new hardware more easily
* It might allow us to build a VST with the exact same engine

For the older hardware, the ability to run on bare metal should be kept of course.

**Reply by diyelectromusic** [reply]
> This means maybe it is worth converting minidexed to use linux instead of bare metal for RPi5

There are already many linux-based synths already and Dexed itself will run on Linux (just look at linuxsynths.com).  I think at this point if this is an area of interest, I'd say just run Zynthian which itself is excellent and includes Dexed as a synth option.

The strength of MiniDexed to me is that it was designed to do one thing and one thing well - an embedded "all in one" environment with a relatively simple user interface that doesn't have to worry about things like being shutdown, working a network stack, handling a graphical UI, and so on.

I really don't see a reason to start to make it into yet another synth engine that runs on Linux, personally...

Kevin

**Reply by probonopd** [reply]
Looks like rsta2 is already working on RPi 5 support in Circle, so I guess we will use that route.

**Comment by stanleygarvey** [comment]
IMHO not a good Idea. Minidexed runs without an OS. A standalone app. Even a minimal distro will have miles of overhead, kernel mods, Alsa and systemd-godforbid...
You can, and I have built dexed on a RPI 4

19 Nov 2023 11:16:28 probonopd ***@***.***>:

**Reply by probonopd** [reply]
Well, what I'd have in mind would be just the kernel plus its modules and one binary. No systemd etc.

**Comment by stanleygarvey** [comment]
Should have said, mainly to have the dexed VST3 for ardour DAW.

19 Nov 2023 17:51:25 Stanley Garvey ***@***.***>:
